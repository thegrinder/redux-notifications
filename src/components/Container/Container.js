import styled from 'styled-components';
import PropTypes from 'prop-types';
import { convertToCssPosition, getFlexDirection } from '../../helpers/utils';

const propTypes = {
  position: PropTypes.array.isRequired,
};

const Container = styled.div`
  position: fixed;
  display: flex;
  z-index: 999999;
  height: 0;
  flex-direction: ${(props) => getFlexDirection(props.position)};
  ${(props) => convertToCssPosition(props.position)}
`;

Container.propTypes = propTypes;

export default Container;
