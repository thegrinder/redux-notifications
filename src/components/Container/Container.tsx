import React, { HTMLAttributes } from 'react';
import {
  convertToCssPosition,
  getFlexDirection,
  Position,
} from '../../helpers/utils';

type ContainerProps = HTMLAttributes<HTMLDivElement> & {
  position: Position;
};

const Container = ({ position, ...rest }: ContainerProps) => (
  <div
    style={{
      position: 'fixed',
      display: 'flex',
      zIndex: 999999,
      height: 0,
      flexDirection: getFlexDirection(position),
      ...convertToCssPosition(position),
    }}
    {...rest}
  />
);

export default Container;
