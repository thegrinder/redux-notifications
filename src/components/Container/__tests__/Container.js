import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Container from '../Container';

const position = ['20px', '30px', 'auto', 'auto'];
const children = <p>test</p>;
const renderComponent = (props = {}) =>
  render(<Container {...props}>{children}</Container>);

describe('<Container />', () => {
  afterEach(cleanup);

  it('should render correctly', () => {
    const {
      container: { firstChild },
    } = renderComponent({ position });
    expect(firstChild).toBeDefined();
    expect(firstChild).toMatchSnapshot();
  });

  it('should have children', () => {
    const {
      container: { firstChild },
      getByText,
    } = renderComponent({ position });
    const childrenElement = getByText('test');
    expect(firstChild).toContainElement(childrenElement);
  });
});
