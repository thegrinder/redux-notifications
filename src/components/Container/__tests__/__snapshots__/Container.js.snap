// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render correctly 1`] = `
<div
  style="position: fixed; display: flex; z-index: 999999; height: 0px; flex-direction: column; top: 20px; right: 30px;"
>
  <p>
    test
  </p>
</div>
`;
